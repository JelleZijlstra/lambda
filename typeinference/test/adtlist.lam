type list a = Nil | Cons a (list a) in
let rec map = \f. \lst. match lst with {
	Nil -> Nil
	| Cons hd tl -> Cons (f hd) (map f tl)
} in
let l = Cons 3 (Cons 1 Nil) in
let l' = map (\x. x = 1) l in
let l'' = map (\x. x + 1) l in
(l', l'')

