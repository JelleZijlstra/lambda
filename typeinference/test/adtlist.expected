%%stdout%%
Result: module
	let __result__ = (((Cons false) ((Cons true) Nil)), ((Cons 4) ((Cons 2) Nil)))
end
%%stderr%%
